oslabs-beta/Kubiquity
# Complete MVP backend

WHAT DOES THIS PR DO?

1. Adds in the finished memory metric endpoint and access to prometheus
2. Fixes duplicate entries showing up in the error table
3. Removed headers from showing up in the error table.

HOW TO TEST THIS PR:

1. Run the following command: `gh pr checkout 17 && npm install` 
2. Make sure to have docker and minikube running.
3. Install helm with `brew install helm`.
4. Add the prometheous repo with `helm repo add prometheus-community https://prometheus-community.github.io/helm-charts`
5. Create a prometheus namespace with `kubectl create namespace prometheus`
6. Install and start Prometheous with `helm upgrade -i prometheus prometheus-community/prometheus --namespace prometheus`
7. Expose the port with `export POD_NAME=$(kubectl get pods --namespace prometheus -l "app=prometheus,component=server" -o jsonpath="{.items[0].metadata.name}") &&  kubectl --namespace prometheus port-forward $POD_NAME 9090`
8. Run `npm run backend`
9. Using Postman or Thunder Client, send a request to `localhost:3000/metrics` and `localhost:3000/errors` to confirm that event logs show up and the memory usage of nodes show up.


ADDITIONAL NOTES:

1. Styling might be a bit off because eslint auto formats but will try to enforce the way we agreed upon.
2. Will want to try to automate exposing the port later on but might be a security risk.
